devito/stencilkernel.py

```diff
@@ -97,7 +98,8 @@ def __init__(self, stencils, **kwargs):
 
         # Introduce C-level profiling infrastructure
         self.sections = OrderedDict()
-        nodes = self._profile_sections(nodes)
+        if self.profiler:
```

Comment:
I'm fine doing it part of this PR. The Julia side is ready and I need  to merge this first then the Julia side.